react/features/app/getRouteToRender.js

```diff
@@ -40,69 +40,128 @@ export type Route = {
+        route.href = locationURL.href;
+
+        return Promise.resolve(route);
+    }
+
+    return getDeepLinkingPage(state)
+        .then(deepLinkComponent => {
+            if (deepLinkComponent) {
+                route.component = deepLinkComponent;
+            } else if (_isSupportedBrowser()) {
                 route.component = Conference;
-                resolve(route);
             } else {
-                // Update the location if it doesn't match. This happens when a
-                // room is joined from the welcome page. The reason for doing
-                // this instead of using the history API is that we want to load
-                // the config.js which takes the room into account.
-                const { locationURL } = state['features/base/connection'];
-
-                // eslint-disable-next-line no-negated-condition
-                if (window.location.href !== locationURL.href) {
-                    route.href = locationURL.href;
-                    resolve(route);
-                } else {
-                    // Maybe show deep-linking, otherwise go to Conference.
-                    getDeepLinkingPage(state).then(component => {
-                        route.component = component || Conference;
-                        resolve(route);
-                    });
-                }
+                route.component = UnsupportedDesktopBrowser;
             }
 
-            return;
+            return route;
+        });
+}
+
+/**
+ * Returns the {@code Route} to display when trying to access the welcome page.
+ *
+ * @param {Object} state - The redux state.
+ * @returns {Promise<Route>}
+ */
+function _getWebWelcomePageRoute(state): Promise<Route> {
+    const route = _getEmptyRoute();
+
+    if (isWelcomePageUserEnabled(state)) {
+        if (_isSupportedBrowser()) {
+            route.component = WelcomePage;
+        } else {
+            route.component = UnsupportedDesktopBrowser;
         }
+    } else {
+        // Web: if the welcome page is disabled, go directly to a random room.
 
-        if (!isWelcomePageUserEnabled(state)) {
-            // Web: if the welcome page is disabled, go directly to a random
-            // room.
+        let href = window.location.href;
 
-            let href = window.location.href;
+        href.endsWith('/') || (href += '/');
+        route.href = href + generateRoomWithoutSeparator();
+    }
 
-            href.endsWith('/') || (href += '/');
-            route.href = href + generateRoomWithoutSeparator();
-        } else if (isWelcomePageAppEnabled(state)) {
-            // Mobile: only go to the welcome page if enabled.
+    return Promise.resolve(route);
+}
 
-            route.component = WelcomePage;
-        }
+/**
+ * Returns whether or not the current browser should allow the app to display.
+ *
+ * @returns {boolean}
+ */
+function _isSupportedBrowser() {
+    if (navigator.product === 'ReactNative') {
+        return false;
```

Comment:
this case is something that should potentially never happen, right ? would it make sense to log an error or throw here ?

Reply:
but then if someone will use this method in other context then probably it would be weird to throw here. Disregard my previous comment